interface/src/app/analyze/analysis/sampleBin.js

```diff
@@ -301,36 +389,50 @@ function($log, $cacheFactory, $q, Sample, Activity, NodeInfo, MathFuncts) {
         return null;
       }
 
-      // (1) we obtain a list of nodes by retrieving node activity
-      //     for the first sample in our volcano plot
-      var firstSampleNodes = this.activityCache.get(sg['group-a'][0]);
-      // (2a) next, we build a new array (`retval`) comprised of `nodeObject`s
-      //      by walking through the `firstSampleNodes` and constructing a
-      //      `nodeObject` for each. [outer .map()]
-      var retval = firstSampleNodes.map(function(val, i, arr) {
-        var mapSampleIdsToActivity = function(sampId, i, arr) {
-          // (2b) the array of activity for each node is built by plucking the
-          //      activity `.value` for each sample within this node from the
-          //      `activityCache` [inner .map()]
-          // FIXME: counting on array order to match node order here
-          return this.activityCache.get(sampId)[val.node - 1].value;
-        };
-        var nodeObject = {
-          'id': val.node, // TODO: map this id to node name (via vega spec?)
-          'activityA': sg['group-a'].map(mapSampleIdsToActivity, this),
-          'activityB': sg['group-b'].map(mapSampleIdsToActivity, this)
-        };
-        nodeObject.diff = (
-          MathFuncts.mean(nodeObject.activityB) -
-          MathFuncts.mean(nodeObject.activityA)
-        );
-        nodeObject.logsig = -Math.log10(MathFuncts.tTest(
-          nodeObject.activityA, nodeObject.activityB
-        ).pValue());
+      // (1a) we obtain a list of nodes by retrieving node activity
+      //      for the first sample in our volcano plot
+      var firstSampleNodes = this.activityCache.get(sg['group-a'][0]).map(
+        function(val) {
+          return val.node;  // extract just the node IDs
+        }
+      );
+      // (1b) now obtain (and cache) a name for each node id
+      var nodeInfoSetPromise = this.getNodeInfoSetPromise(firstSampleNodes);
+      var mapNodesToNodeInfo = function() {
+        // (2a) next, we build an array (replacing `volcanoData.source`)
+        //      comprised of `nodeObject`s by walking through the
+        //      `firstSampleNodes` and constructing a `nodeObject` for
+        //      each. [outer .map()]
+        cbSampleBin.volcanoData.source = firstSampleNodes.map(function(id) {
+          var mapSampleIdsToActivity = function(sampId) {
+            // (2b) the array of activity for each node is built by plucking the
+            //      activity `.value` for each sample within this node from the
+            //      `activityCache` [inner .map()]
+            // FIXME: counting on array order to match node order here
```

Comment:
I have a bunch of FIXME and TODO comments throughout the code that I do intend to return to. The way I use them, FIXME calls attention to something that I know isn't quite right but cannot be addressed fully at the moment while TODO is helpful to future me (or someone else) to guide future development steps. Ideally, there would be no FIXMEs in production, but I view TODOs as helpful comments.

The collection of FIXMEs in `analysis.js`, `sampleBin.js` and `view.js`, in particular, reflect work that I am currently doing to finish off the basic volcano plot features.

I have a command in my text editor that scans the source tree (or an individual file) and lists any FIXME or TODO along with the comment on the rest of that line. It's a very helpful index I use for quickly returning to something that's work in progress.

There are a handful of older FIXME items that might qualify for your "won't actually get fixed" observation, but most have corresponding issues and we are tracking them that way. I think it's helpful in those cases to leave the comments as guideposts for the eventual fixer.

That said, if others find this sort of comment distracting or objectionable we should maybe come up with a GreeneLab policy to include in our code review checklist. I know there is a line in our `codeclimate.yml` we can configure to have it gripe at us about FIXME, for example.